File: backend/inventory-service/src/main/java/com/storelense/inventory/dto/GateCheckRequest.java
package com.storelense.inventory.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.List;
import java.util.UUID;

public record GateCheckRequest(
        @NotNull  UUID         storeId,
        UUID                   guardUserId,
        String                 billRef,
        int                    expectedCount,
        int                    matchedCount,
        int                    extraCount,
        @NotBlank String       outcome,
        List<String>           epcsMatched,
        List<String>           epcsExtra
) {}

File: android-xr22-gate/app/src/main/java/com/storelense/c66/data/remote/dto/GateCheckRequest.kt
package com.storelense.c66.data.remote.dto

data class GateCheckRequest(
    val storeId:       String,
    val billRef:       String,
    val expectedCount: Int,
    val matchedCount:  Int,
    val extraCount:    Int,
    val outcome:       String,   // "RELEASED" | "FLAGGED" | "ABANDONED"
    val epcsMatched:   List<String>,
    val epcsExtra:     List<String>
)
